labs/lab-12/tasks/feeling-chained/solution/buff-ovf3.c

```diff
@@ -77,8 +77,21 @@ void gateway(void)
 	fgets(buff1, 300, stdin);
 }
 
+/* ROP gadgets for x86-64 */
+void gadget_pop_rdi(void)
+{
+	__asm__("pop %rdi; ret");
+}
+
+void gadget_pop_rsi(void)
+{
+	__asm__("pop %rsi; ret");
+}
```

Comment:
You generally find `pop rdi; ret` and `pop rsi; ret` gadgets in regular binaries. Why do you need these here?